src/map/lapack2flash/FLASH_lapack2flash_util.c

```diff
@@ -245,53 +241,61 @@ int FLAME_QR_piv_preorder( FLA_Obj A, int *jpiv_lapack, int *jpiv_fla )
 }
 */
 
-/*---------------------Get/Set Helper---------------------*/
-
-FLA_Error FLASH_set_preferred_blocksize( dim_t blocksize )
-{
-    flash_blocksize = blocksize;
-
-    return FLA_SUCCESS;
-}
-
-dim_t FLASH_get_preferred_blocksize( void )
-{
-    return flash_blocksize;
-}
-
-FLA_Error FLASH_set_n_preferred_threads( unsigned int threads )
+FLA_Bool FLASH_Check_Offload_to_GPU( dim_t blocksize, int m, int n, unsigned int offset )
 {
-    flash_n_threads = threads;
-
     #ifdef FLA_ENABLE_HIP
-        int device_count = 1;
-        FLASH_Queue_available_devices_hip( &device_count );
-
-        // Pass number of threads unless it's more than available devices
-        FLASH_Queue_set_num_threads( min( threads, device_count) );
-    #else
-        FLASH_Queue_set_num_threads( threads );
-    #endif
-
-    return FLA_SUCCESS;
-}
+        // 'offset' is the number of tiles where it crosses over from CPU to GPU offloading
+        // Checking to see if the area of m and n is more than X tiles. Ex, blocksize = 10, m = 30, n =30  (9 tiles)
+        // Checking if m is more than X tiles long. Ex, blocksize = 10, m = 41, n =1 (5 tiles)
+        // Checking if n is more than X tiles long. Ex, blocksize = 10, m = 1, n =41 (5 tiles)
+        // If it is less than X tiles (starting with 4 tiles) then do not off load
+        if ( blocksize * blocksize * offset < m * n ||
+             blocksize             * offset < m     ||
+             blocksize             * offset <     n )
+        {
+            if (!FLASH_Queue_get_enabled_hip())
+            {
+                // If hip is not enabled, enable it
+                FLASH_Queue_enable_hip();
 
-unsigned int FLASH_get_n_preferred_threads( void )
-{
-    // Returning the number of preferred threads, not actual number.
-    return flash_n_threads;
-}
+                // Return 1 saying we need to flip this back after the FLASH_ call
+                return TRUE;
+            }
+        }
+        else
+        {
+            if (FLASH_Queue_get_enabled_hip())
+            {
+                // If hip is enabled, disable it
+                FLASH_Queue_disable_hip();
 
-FLA_Error FLASH_set_depth( dim_t depth )
-{
-    flash_depth = depth;
+                // Return 1 saying we need to flip this back after the FLASH_ call
+                return TRUE;
+            }
+        }
+    #endif
 
-    return FLA_SUCCESS;
+    // Return 0 saying we doing need to flip this back
+    return FALSE;
 }
 
-dim_t FLASH_get_depth( void )
+void FLASH_Toggle_GPU_Offload( FLA_Bool toggle )
```

Comment:
FLASH_Check_Offload_to_GPU() returns a 1 it changed the hip state. This will change the hip state back if passed that 1